scripts/dts/edtsdatabase.py

```diff
@@ -0,0 +1,193 @@
+#
+# Copyright (c) 2018 Bobby Noelte
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from collections.abc import Mapping
+import json
+
+##
+# @brief Extended DTS database
+#
+# Database schema:
+#
+# _edts dict(
+#   'devices':  dict(device-id :  device-struct),
+#   'compatibles':  dict(compatible : sorted list(device-id)),
+#   'device-types': dict(device-type : sorted list(device-id)),
+#   ...
+# )
+#
+# device-struct: dict(
+#   'device-id' : device-id,
+#   'device-type' : list(device-type) or device-type,
+#   'compatible' : list(compatible) or compatible,
+#   'label' : label,
+#   property-name : property-value ...
+# )
+#
+# Database types:
+#
+# device-id: <compatible>_<base address in hex>_<offset in hex>,
+# compatible: any of ['st,stm32-spi-fifo', ...] - 'compatibe' from <binding>.yaml
+# device-type: any of ['GPIO', 'SPI', 'CAN', ...] - 'id' from <binding>.yaml
+# label: any of ['UART_0', 'SPI_11', ...] - label directive from DTS
+#
+class EDTSDatabase(Mapping):
```

Comment:
For clarity, it would be nice to split between producer and consumer functions.

Reply:
> For clarity, it would be nice to split between producer and consumer functions.

I tried to by the function names:

  - Producer
    - insert_device_property
    - save
  - Consumer
    - device_property
    - load

I´m open to better method names.

Reply:
So a clear split in the file would be good. get_... for consumer could be nice as well
What about sub classes?

Reply:
> What about sub classes?

OK will go for mixin classes like codegen uses to separate the different functions.